### PR TITLE
⚠️ Decouple plugin requirements from Config interface

### PR DESCRIPTION
**Pull request by Adirio**
Extract the single version per config file out of config as it is a plugin restriction, not a project configuration file restriction.

Closes: #2046

### CODE REVIEW

File: pkg/plugin/util/helpers.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+// IsOnlyWebhookVersion returns true if webhookVersion is the first or the only existing webhook version in the project.
+func IsOnlyWebhookVersion(config config.Config, webhookVersion string) bool {
+	return isOnlyAPIVersion(config.ListWebhookVersions(), webhookVersion)
+}
+
+func isOnlyAPIVersion(versions []string, version string) bool {
+	return len(versions) == 0 || (len(versions) == 1 && versions[0] == version)
+}
```
Comment (camilamacedo86) [inline pkg/plugin/util/helpers.go:35]: I understand that you change that for a List of webhook versions and List of CRD versions to allow the possibility of the tool works with many versions in the future. 

However, I do not think that we should have IsOnly funcs here. We would still have isWebhoohVersionCompatible  and IsCRDVersionCompatible where its responsibility would ensure that the version informed can or not be used.

File: pkg/plugins/golang/v3/api.go [hunk 2]
```diff
@@ -189,7 +190,7 @@ func (p *createAPISubcommand) Validate() error {
 		}
 
 		// Check CRDVersion against all other CRDVersions in p.config for compatibility.
-		if !p.config.IsCRDVersionCompatible(p.resource.API.CRDVersion) {
+		if !pluginutil.IsOnlyCRDVersion(p.config, p.resource.API.CRDVersion) {
```
Comment (camilamacedo86) [inline pkg/plugins/golang/v3/api.go:193]: I do not think that this change is nice. I did not like the idea of have this funcs IsOnlyCRDVersion. The previous one represents a better need. We will pass the version of CRD or Webhook which we want to use and check if the project config allows or not it.

File: pkg/plugin/util/helpers.go (hunk 1 above)
Comment (Adirio) [reply pkg/plugin/util/helpers.go:35]: We talked about this and `hasDifferentAPIVersion` seems like a better name (for the negated function).

**Review by camilamacedo86 — COMMENTED** [review]
I agree with this change. Note that the config is responsible to have funcs to return its values, however, the rule to allow or not different APIs is specific to the plugin since we could, for example, support it for go/v4.

In this way, since it is a very small and concise change that does not affect the end-user behaviours and is only more a cleanup. I am happy in 

/lgtm

**Pull request merged**